auth_backends/backends.py

```diff
@@ -243,61 +242,14 @@ def _to_language(locale):
 class EdXOAuth2(EdXBackendMixin, BaseOAuth2):
     name = 'edx-oauth2'
 
-    def endpoint(self):
-        return self.setting('ENDPOINT').strip('/')
-
-    def provider_configuration(self):
-        cache_key = 'edx_oauth2_provider_configuration'
-        config = cache.get(cache_key)
-
-        if not config:
-            config = self.get_json(self.endpoint() + '/.well-known/openid-configuration')
-
-            # Cache for one week since the configuration rarely changes
-            cache.set(cache_key, config, self.setting('PROVIDER_CONFIGURATION_CACHE_TTL', 604800))
-
-        return config
-
     def authorization_url(self):
-        return self.provider_configuration().get('authorization_endpoint')
+        return self.setting('AUTHORIZATION_URL')
```

Comment:
Is there a reason the implementation of these `url` methods need to be different from their corresponding implementations in `EdXOpenIdConnect`?

Reply:
I'm not really sure how the `url` related properties are working in `EdXOpenIdConnect`. The methods defined on this class override the implementations in the class hierarchy using the `setting` method to extract values from Django settings. I think eventually `EdXOpenIdConnect` is going away, so it is probably better to avoid coupling the implementation of these methods between this class and that one.

Reply:
Ok. But does it make sense to have the client just configure the `JWT_PROVIDER_URL_ROOT` and then all 3 of these can be based on that single config value?